src/Microsoft.VisualBasic/src/Microsoft/VisualBasic/FileIO/SpecialDirectories.vb

```diff
@@ -0,0 +1,244 @@
+        ''' </summary>
+        ''' <value>A String containing the path to the current user's Desktop directory.</value>
+        ''' <remarks>This directory is C:\Document and Settings\[UserName]\Desktop.</remarks>
+        Public Shared ReadOnly Property Desktop() As String
+            Get
+                Return GetDirectoryPath(Environment.GetFolderPath(SpecialFolder.Desktop), SR.IO_SpecialDirectory_Desktop)
+            End Get
+        End Property
+
+        ''' <summary>
+        ''' Returns the directory used to store program shortcuts from Start Menu for current user.
+        ''' </summary>
+        ''' <value>A String containing the path to the Start Menu \ Programs directory.</value>
+        ''' <remarks>This directory is C:\Document and Settings\[UserName]\Start Menu\Programs.</remarks>
+        Public Shared ReadOnly Property Programs() As String
+            Get
+                Return GetDirectoryPath(Environment.GetFolderPath(SpecialFolder.Programs), SR.IO_SpecialDirectory_Programs)
+            End Get
+        End Property
+
+        ''' <summary>
+        ''' Return the program files directory.
+        ''' </summary>
+        ''' <value>A String containing the path to the default program directories.</value>
+        ''' <remarks>This directory is C:\Program Files.</remarks>
+        Public Shared ReadOnly Property ProgramFiles() As String
+            Get
+                Return GetDirectoryPath(Environment.GetFolderPath(SpecialFolder.ProgramFiles), SR.IO_SpecialDirectory_ProgramFiles)
+            End Get
+        End Property
+
+        ''' <summary>
+        ''' Return the directory that contain temporary files for the current user.
+        ''' </summary>
+        ''' <value>A String containing the path to the temporary directory for the current user.</value>
+        ''' <remarks>
+        ''' According to Win32 API document, GetTempPath should always return a value even if TEMP and TMP = "".
+        ''' Also, this is not updated if TEMP or TMP is changed in Windows. The reason is
+        '''     each process has its own copy of the environment variables and this copy is not updated.
+        ''' </remarks>
+        Public Shared ReadOnly Property Temp() As String
+            Get
+                Return GetDirectoryPath(IO.Path.GetTempPath(), SR.IO_SpecialDirectory_Temp)
+            End Get
+        End Property
+
+        ''' <summary>
+        ''' Returns the directory that serves as a common repository for data files
+        ''' from your application used only by the current user.
+        ''' </summary>
+        ''' <value>A String containing the path to the directory your application can use to store data for the current user.</value>
+        ''' <remarks>
+        ''' We chose to use UserAppDataPath instead of LocalUserAppDataPath since this directory
+        ''' will work with Roaming User as well.
+        ''' </remarks>
+        Public Shared ReadOnly Property CurrentUserApplicationData() As String
```

Comment:
>CurrentUserApplicationData [](start = 40, length = 26)

Should this function simply `Throw New PlatformNotSupportedException()` on non-Windows platforms?

Reply:
I don't know I just past through what I get from Environment for the base and then add separation the way Windows.Forms does and for the most part it work except on Nano and one other Windows platform. I am not sure what I get on Linux make sense since I don't know Linux. But this is a heavily used API in my experience and application need a place to write. Also I am sure the way I am creating separation does not match the current implementation, 1/2 my commits are just trying to understand what I get back and what should I get be returning. I have a parallel test on .Net Framework and both CurrentUserApplicationData and AllUsersApplicationData never match on my current Windows 10 desktop.